create_session_parquet_files.py

```diff
@@ -1,60 +1,58 @@
-import os
 import argparse
-import re
+from pathlib import Path
+from collections import defaultdict
 
 from astropy.table import Table, vstack
-import pandas as pd
-import numpy as np
-
-# regular expression to match the session name given a path to antenna FITS file
-SESSION_REGEX = re.compile(r"/[^/]*/Antenna")
-
-def group_files_df(manifest_path: str):
-    """Returns a pd dataframe with one column containing all the paths listed in the manifest file, and
-    another column containing each file's associated session"""
-    lines = open(manifest_path, "r").readlines()
-    df = pd.DataFrame(lines, columns=["path"])
-    df["path"] = df["path"].map(lambda x: x.replace("\n", ""))
-    # get session name
-    df["session"] = df["path"].map(lambda x: re.search(SESSION_REGEX, x).group(0).split("/")[1])
-    return df
-
-
-def create_session_antenna_df(df, session):
-    """Generates a dataframe containing the antenna position data for a given session"""
-    sliced = df[df["session"] == session]
+
+
+def group_files(manifest_path: Path):
+    """Generates a dictionary grouping FITS file paths with their associated GBT session"""
+    
+    d: defaultdict[str, list[Path]] = defaultdict(list)
+    with open(manifest_path, "r") as file:
+        {d[Path(path).parent.parent.name].append(Path(path.rstrip("\n"))) for path in file}   
```

Comment:
While this does work, I typically avoid using a comprehension if I'm not going to use its return value

```suggestion
         for path in file:
             d[Path(path).parent.parent.name].append(Path(path.rstrip("\n")))
```